src/main/java/org/aksw/simba/squirrel/metadata/CrawlingActivity.java

```diff
@@ -112,23 +131,85 @@ private void countTriples() {
         }
     }
 
+    /**
+     *
+     * @return unique Id
+     */
+
     public String getId() {
         return id;
     }
 
-    public String getDateStarted() {
-        String dateString = dateStarted.toString();
-        dateString = dateString.replace(" ", "_");
-        dateString = dateString.replace(":", "_");
-        return dateString;
+
+    public String getHost() {
+        try {
+            WorkerConfiguration workerConfiguration = WorkerConfiguration.getWorkerConfiguration();
+            String httpPrefix = "http://localhost:" + workerConfiguration.getSqarqlPort() + "/";
+            return httpPrefix;
+        } catch (Exception e) {
+            return null;
+        }
+    }
+
+    private Map<String,Object> addStep()
+    {
+        int count = 1;
+        if (worker instanceof WorkerImpl) {
+            List<Field> list = listAllFields(worker);
+
+            for (Field field : list)
+            {
+                String l = "step" + count++;
+                uri.addData(l,field.getClass().getSimpleName());
+            }
+
+
+        }
+        return uri.getData();
+    }
+
+    private List<Field> listAllFields(Object k)
+    {
+        List<Field> fields = new ArrayList<>();
+        Class tmpclass = k.getClass();
+        while(tmpclass!=null)
+        {
+            fields.addAll(Arrays.asList(tmpclass.getDeclaredFields()));
+        }
+        return fields;
+
     }
 
-    public String getDateEnded() {
+    public String getHadPlan()
+    {
+        ArrayList<String> list = new ArrayList<>();
+        for(Object o: addStep().values())
+        {
+            list.add(o.toString());
+
+        }
+        return list.toString();
+    }
+
+
+    public LocalDateTime getLocalDateTime()
+    {
+
+        LocalDateTime formatter = LocalDateTime.now();
+        return formatter;
+    }
+
+
+
+    public CrawleableUri getCrawleableUri()
+    {
+        return uri;
+    }
+
+    public String getGraphId(CrawleableUri uri)
+    {
+        return graphId;
 
-        String dateString = dateEnded.toString();
-        dateString = dateString.replace(" ", "_");
-        dateString = dateString.replace(":", "_");
-        return dateString;
     }
 
     public int getNumTriples() {
```

Comment:
Your class should provide setters for the attributes. At least those attributes which are not available during the construction of the CrawlingActivity object and have to be set later. At the moment, the number of triples is always 0.